lib/teleterm/gateway/gateway.go

```diff
@@ -103,6 +113,13 @@ func New(cfg Config) (*Gateway, error) {
 		localProxy:   localProxy,
 	}
 
+	if cfg.OnExpiredCert != nil {
+		localProxyMiddleware.onExpiredCert = func(ctx context.Context) error {
+			err := cfg.OnExpiredCert(ctx, gateway)
+			return trace.Wrap(err)
+		}
+	}
```

Comment:
It's kind of an awkward way to initialize the middleware and add the callback later. However, I want to be able to pass `gateway.Gateway` as the argument so that's why I first instantiate the middleware, then the gateway and then set the callback.